File: Complement of Base 10 Integer/solution.js
/**
 * @param {number} n
 * @return {number}
 */
var bitwiseComplement = function (n) {
    if (n === 0) return 1;
    // 5 returns [1, 0, 1]
    const getBinary = num => {
        let result = [];
        while (num > 0) {
            result.push(num % 2);
            num = Math.floor(num / 2);
        }
        result.reverse();
        return result;
    }
    // [1, 0, 1] returns [0, 1, 0]
    const getComplement = arr => {
        for (let i = 0; i < arr.length; i++) {
            arr[i] = arr[i] === 1 ? 0 : 1;
        }
        return arr;
    }
    // [0, 1, 0] returns [1, 0, 1]
    const getDecimal = arr => {
        let result = 0;
        for (let i = 0; i < arr.length; i++) {
            result += arr[i] === 1 ? Math.pow(2, arr.length - i - 1) : 0;
        }
        return result;
    }
    return getDecimal(getComplement(getBinary(n)))
};

File: Complement of Base 10 Integer/solution.py
import math
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0: return 1
        def getBinary(num):
            # 5 returns [1, 0, 1]
            result = []
            while num > 0:
                result.append(num%2)
                num = num // 2
            result.reverse()
            return result
        
        def getDecimal(num_arr):
            # [0, 1, 0] returns 2
            result = 0
            for i, num in enumerate(num_arr):
                result += math.pow(2, len(num_arr)-i-1) if num == 1 else 0
            return result
        
        def complementArr(arr):
            # [1, 0, 1] return [0, 1, 0]
            for i in range(len(arr)):
                arr[i] = 0 if arr[i] == 1 else 1
            return arr
                
        
        return int(getDecimal(complementArr(getBinary(n))))
        